From 65f101b1fc282b05bf797b094e6face9bf66b457 Mon Sep 17 00:00:00 2001
From: Miroslaw Swiader <miroslaw.swiader@intel.com>
Date: Tue, 9 Feb 2016 17:47:17 +0100
Subject: [PATCH 2/5] usb: dwc3: set LPM_NYET_treshold to 0

This is workarround for increase performance on USB2 transfers
with adb protocol.

This patch is set LPM_NYET_treshold to 0 value when LPM Errata is enabled
to increase performance with adb.

Change-Id: I09b29b3e43687b3fc1f741713cd128fb1e4c58d5
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-19751
Signed-off-by: Miroslaw Swiader <miroslaw.swiader@intel.com>
Reviewed-on: https://android.intel.com:443/468861
---
 drivers/usb/dwc3/core.h   | 1 +
 drivers/usb/dwc3/gadget.c | 8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/drivers/usb/dwc3/core.h b/drivers/usb/dwc3/core.h
index f1c0a1c..a433082 100644
--- a/drivers/usb/dwc3/core.h
+++ b/drivers/usb/dwc3/core.h
@@ -214,6 +214,7 @@
 #define DWC3_DCTL_RUN_STOP	(1 << 31)
 #define DWC3_DCTL_CSFTRST	(1 << 30)
 #define DWC3_DCTL_LSFTRST	(1 << 29)
+#define DWC3_DCTL_L1_NYET_TRESH	(0xF << 20)
 
 #define DWC3_DCTL_HIRD_THRES_MASK	(0x1f << 24)
 #define DWC3_DCTL_HIRD_THRES(n)	((n) << 24)
diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index fbb05c5..b152c4b 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -2523,6 +2523,14 @@ static void dwc3_gadget_conndone_interrupt(struct dwc3 *dwc)
 		 */
 		reg |= DWC3_DCTL_HIRD_THRES(12);
 
+		/*
+		* WORKARROUND: When LPM Errata is enabled LPM_NYET_treshold
+		* is set to 0 to improve performance on USB2
+		* when LPM Errata is enabled
+		*/
+		if ((reg & DWC3_DCTL_L1_NYET_TRESH) == DWC3_DCTL_L1_NYET_TRESH)
+			reg &= ~DWC3_DCTL_L1_NYET_TRESH;
+
 		dwc3_writel(dwc->regs, DWC3_DCTL, reg);
 	}
 
-- 
1.9.1

